src/tigerbeetle/repl.zig

```diff
@@ -0,0 +1,813 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const assert = std.debug.assert;
+
+const constants = @import("../constants.zig");
+const stdx = @import("../stdx.zig");
+
+const build_options = @import("vsr_options");
+
+const vsr = @import("vsr");
+const IO = vsr.io.IO;
+const Storage = vsr.storage.Storage;
+const StateMachine = vsr.state_machine.StateMachineType(Storage, constants.state_machine_config);
+const MessagePool = vsr.message_pool.MessagePool;
+
+const tb = vsr.tigerbeetle;
+
+fn print(comptime fmt: []const u8, args: anytype) !void {
+    const stdout = std.io.getStdOut().writer();
+    try stdout.print(fmt, args);
+}
+
+pub const Parse = struct {
+    input: []const u8,
+    offset: usize = 0,
+
+    pub const Operation = enum {
+        none,
+        help,
+        create_accounts,
+        create_transfers,
+        lookup_accounts,
+        lookup_transfers,
+    };
+
+    pub const LookupSyntaxTree = struct {
+        id: u128,
+    };
+
+    pub const ObjectSyntaxTree = union(enum) {
+        account: tb.Account,
+        transfer: tb.Transfer,
+        id: LookupSyntaxTree,
+    };
+
+    pub const StatementSyntaxTree = struct {
+        operation: Operation,
+        // Must be an array of pointers to structs because
+        // there is a Zig bug that causes segfaults sometimes
+        // when expanding capacity for an array of structs.
+        // https://github.com/ziglang/zig/issues/8655
+        args: []*ObjectSyntaxTree,
+    };
+
+    fn fail_at(
+        parse: *Parse,
+        comptime fmt: []const u8,
+        args: anytype,
+    ) !void {
```

Comment:
Yes, format strings are always comptime. But splitting these would mean that `.fail_at` no longer needs to be specialized, since `stderr.print` is being specialized directly already.